src/oracle/middleware.js

```diff
@@ -0,0 +1,761 @@
+      let latestRound = await queryRound(jobName);
+
+      //get latest submitted round
+      var latestSubmittedRound = await getLatestSubmittedRound()
+    
+      //update jobs table
+      await updateTable("jobs", {"last_result" : latestPrice, "last_reported_round": latestSubmittedRound}, jobName)
+
+      //update rounds table
+      await updateTable("rounds", latestRound, jobName)
+
+      //if latest round is bigger than last reported round
+      if (latestRound.round_id > latestSubmittedRound) {
+        //if submitted, update last_reported_round
+        if (latestRound.submission_made) {
+          await updateTable("jobs", {"last_reported_round": latestSubmittedRound}, jobName)
+        }
+        //if not found send job request
+        else {
+          console.log("Found new round.")
+          sendRequest = 3
+        }
+      }
+
+      //if there's a price deviation
+      let priceDev = Math.abs((latestPrice - currentPrice) / currentPrice) * 100
+
+      if (priceDev > 0){
+        console.log("Found a price deviation for", jobName, "of", priceDev, "%. Latest price:", latestPrice, " Current Price:", currentPrice)
+      }
+
+      //get feed deviation percentage threshold
+      let priceDeviationPercentage = feeds[jobName].priceDeviationPerc
+      if (priceDev > priceDeviationPercentage) {
+        sendRequest = 2
+      }
+
+      //if there is a request to be sent
+      if (sendRequest != 0) {
+
+        //get seconds now
+        let secondsNow = Date.now() / 1000
+        //check seconds passed from last request
+        let query = await queryTable("jobs", ["last_request_sent"], jobName)
+        let secondsPassed = secondsNow - query.last_request_sent
+
+        //check if allowed to send - 12 seconds passed or a request has not been made.
+        //12 seconds chosen to wait at least 2 blocks
+        query = await queryTable("jobs", ["last_received_request_id", "request_id"], jobName)
+        let allowedSend = query.request_id == query.last_received_request_id || secondsPassed > Number(SEND_CHECK_INTERVAL)
+
+        //if a request has not been made yet
+        if (allowedSend) {
+          //submit job
+          console.log("Initialising new CL job request")
+          submitNewJob(jobName, sendRequest)
+        }
+        else{
+          console.log("Will not be initialising new job request - Still waiting for request", query.request_id, "to finish. Last finished request is", query.last_received_request_id)
+        }
+      }
+    }
+  }, priceQueryInterval * 1_000);
+
+}
+
+/**
+  * Function to create a bridge which listens from the Chainlink node for
+  * new jobs, job removals and job run results
+  * @param {*} PORT the port to listen on
+  */
+const startBridge = (PORT) => {
+
+  console.log("Bridge started")
+  const app = express();
+  app.use(bodyParser.json());
+
+  /**
+  * POST /adapter endpoint
+  * This is used to listen for job run results
+  */
+  app.post('/adapter', async (req, res) => {
```

Comment:
Is it possible to add a link of what is expected to be received?

Reply:
There is no link offering such details. Only this but it does not specify details.
https://docs.chain.link/chainlink-nodes/external-initiators/external-initiators-in-nodes